src/net.rs

```diff
@@ -0,0 +1,50 @@
+use async_trait::async_trait;
+use anyhow::Error;
+use log::debug;
+use rand::Rng;
+
+#[async_trait]
+pub trait Retry {
+    async fn auto_retry(&self, attempts: u64) -> Result<reqwest::Response, Error>;
+}
+
+#[async_trait]
+impl Retry for reqwest::RequestBuilder {
```

Comment:
Ok, I don't understand how this gets used, wouldn't we have to be calling auto_retry on a request builder somewhere? but we're not?

Reply:
Oops, that's a leftover from a previous patch. My bad.